stor/dx.py

```diff
@@ -436,17 +816,19 @@ def canonical_project(self):
         """
         if utils.is_valid_dxid(self.project, 'project'):
             return self.project
-        else:
-            try:
-                proj_dict = dxpy.find_one_project(
-                    name=self.project, level='VIEW', zero_ok=True, more_ok=False)
-                if proj_dict is None:
-                    raise ProjectNotFoundError('No projects were found with given name ({})'
-                                               .format(self.project))
-                return proj_dict['id']
-            except DXError as e:
-                raise DuplicateProjectError('Duplicate projects were found with given name ({})'
-                                            .format(self.project), e)
+
+        try:
+            proj_dict = dxpy.find_one_project(
```

Comment:
can you make sure you have a test for this when auth token is not set? (i.e., when there's an auth error) - just helpful to understand what'll happen in that case

Reply:
It just assumes the user is anonymous, and says there's no projects found. Closing this.